src/art.rs

```diff
@@ -788,7 +798,21 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
         // update the existing value in the Twig node.
         if let NodeType::Twig(ref mut twig) = &mut cur_node.node_type {
             if is_prefix_match && cur_node_prefix.len() == key_prefix.len() {
-                twig.insert_mut(value.clone(), commit_version, ts);
+                if replace {
+                    if commit_version > twig.version {
```

Comment:
Can we pass down this logic to the twig node, and call it `replace_if_newer` or something similar because it is logic specific to insert within a twig node?

Reply:
Added `TwigNode::replace_if_newer_mut`.